sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -448,14 +448,6 @@ def test_count_nonzero():
     assert (count_nonzero(X_csr, axis=0, sample_weight=sample_weight).dtype ==
             count_nonzero(X_csr, axis=1, sample_weight=sample_weight).dtype)
 
-    # Check dtypes with large sparse matrices too
```

Comment:
Better than this (but perhaps still not ideal) would be to have this all in a try-except block, catching and passing if the error message contains "according to the rule 'safe'" and if `np.intp().nbytes < 8`.

Reply:
This LGTM from my side and I can't figure out a better solution.